packages/react/src/components/provider.tsx

```diff
@@ -0,0 +1,44 @@
+import { useState, useMemo, useEffect, type ReactNode } from 'react';
+import { PenumbraProviderNotAvailableError, PenumbraSymbol } from '@penumbra-zone/client';
+import { penumbraContext } from '../hooks/penumbra';
+
+export const PenumbraProvider = ({
+  children,
+  providerOrigin,
+  requestApproval = false,
+  required = false,
+}: {
+  children?: ReactNode | ReactNode[];
+  providerOrigin: string;
+  requestApproval?: boolean;
+  required?: boolean;
+}) => {
+  console.log('PenumbraProvider', { children, providerOrigin, requestApproval, required });
+  const provider = useMemo(() => window[PenumbraSymbol]?.[providerOrigin], [providerOrigin]);
+
+  const [requested, setRequested] = useState(false);
+  const [providerConnected, setProviderConnected] = useState<boolean>();
+  const [port, setPort] = useState<MessagePort>();
+
+  setProviderConnected(provider?.isConnected());
```

Comment:
suggestion: change `providerConnected` to be a memo instead of state since it depends on the `provider` value and doesn't run `setProviderConnected` elsewhere

Reply:
the return value of isConnected will change, without any change in object ref of provider. wrapping in useMemo would fail to update after the first call